src/libsync/syncfilestatustracker.cpp

```diff
@@ -140,22 +127,76 @@ void SyncFileStatusTracker::slotPathTouched(const QString& fileName)
+    int count = _syncCount[relativePath]++;
+    if (!count) {
+        if (emitStatusChange)
+            emit fileStatusChanged(getSystemDestination(relativePath), fileStatus(relativePath));
+
+        // We passed from OK to SYNC, increment the parent to keep it marked as
+        // SYNC while we propagate ourselves and our own children.
+        Q_ASSERT(!relativePath.endsWith('/'));
+        int lastSlashIndex = relativePath.lastIndexOf('/');
+        if (lastSlashIndex != -1)
+            incSyncCount(relativePath.left(lastSlashIndex), EmitStatusChange);
+        else if (!relativePath.isEmpty())
+            incSyncCount(QString(), EmitStatusChange);
+    }
+}
+
+void SyncFileStatusTracker::decSyncCount(const QString &relativePath, EmitStatusChangeFlag emitStatusChange)
+{
+    int count = --_syncCount[relativePath];
+    if (!count) {
+        // Remove from the map, same as 0
+        _syncCount.remove(relativePath);
+
+        if (emitStatusChange)
+            emit fileStatusChanged(getSystemDestination(relativePath), fileStatus(relativePath));
+
+        // We passed from SYNC to OK, decrement our parent.
+        Q_ASSERT(!relativePath.endsWith('/'));
+        int lastSlashIndex = relativePath.lastIndexOf('/');
+        if (lastSlashIndex != -1)
+            decSyncCount(relativePath.left(lastSlashIndex), EmitStatusChange);
+        else if (!relativePath.isEmpty())
+            decSyncCount(QString(), EmitStatusChange);
+    }
+}
+
 void SyncFileStatusTracker::slotAboutToPropagate(SyncFileItemVector& items)
 {
+    // New sync, clear the sync counts to reduce the impact of a bug that would cause unsymetrical inc/dec
+    Q_ASSERT(_syncCount.isEmpty());
+    _syncCount.clear();
+
     std::map<QString, SyncFileStatus::SyncFileStatusTag> oldProblems;
     std::swap(_syncProblems, oldProblems);
 
     foreach (const SyncFileItemPtr &item, items) {
-        // qDebug() << Q_FUNC_INFO << "Investigating" << item->destination() << item->_status;
+        // qDebug() << Q_FUNC_INFO << "Investigating" << item->destination() << item->_status << item->_instruction;
 
         if (showErrorInSocketApi(*item)) {
             _syncProblems[item->_file] = SyncFileStatus::StatusError;
             invalidateParentPaths(item->destination());
         } else if (showWarningInSocketApi(*item)) {
             _syncProblems[item->_file] = SyncFileStatus::StatusWarning;
         }
+
+        // Mark this path as syncing for instructions that will result in propagation,
+        // but DontEmitStatusChange since we're going to emit for ourselves using the
+        // info in the SyncFileItem we received, parents will still be emit if needed.
+        if (item->_instruction != CSYNC_INSTRUCTION_NONE
+            && item->_instruction != CSYNC_INSTRUCTION_UPDATE_METADATA
+            && item->_instruction != CSYNC_INSTRUCTION_IGNORE
+            && item->_instruction != CSYNC_INSTRUCTION_ERROR) {
```

Comment:
Humm not sure, I think that "should count as progress" is a different requirement than "should be shown as syncing in the overlay icons". Moreover, the overlay icon situation is also restricted by what can give us a symetrical inc/dec, so I wouldn't want a change in the progress logic to inadvertently affect the status tracker. It's also not a big deal if they diverge in the future, so I would only see saving a few similar lines of code as a benefit.